doc/OnlineDocs/principles.rst

```diff
@@ -0,0 +1,193 @@
+Development Principles
+======================
+
+The Pyomo development team follows a set of development principles.
+In order to promote overall transparency, this page is intended to document
+those principles, to the best of our ability, for users and potential
+contributors alike. Please also review Pyomo's recent publication
+on the history of its development for a holistic view into the changes
+of these principles over time [MHJ+25]_.
+
+Backwards Compatibility
+-----------------------
+
+Commitment to Published Interfaces
+++++++++++++++++++++++++++++++++++
+
+If functionality is published in the most recent edition of the
+Pyomo book [PyomoBookIII]_ and corresponding examples, we treat that as
+a public API commitment. Once published in the book, those APIs will not
+change until the next major Pyomo release, which will coincide with a
+new edition of the book.
+
+This commitment ensures that teaching materials, training resources, and
+long-term codebases built on those examples will remain valid across an
+entire major release.
+
+Stable, Supported APIs
+++++++++++++++++++++++
+
+Functionality that is part of the Pyomo source tree but not explicitly
+included in the book is also expected to be stable if it resides outside
+the ``contrib`` (or future ``addons`` / ``devel``) directories. This is
```

Comment:
should this just say "``devel``" instead of "``addons`` / ``devel``"? I think the contribution guide stated that `addons` was supposed to be stable as well.